src/neo/IO/Json/JPathToken.cs

```diff
@@ -0,0 +1,308 @@
+                    break;
+                default:
+                    throw new FormatException();
+            }
+        }
+
+        private static void ProcessBracket(ref JObject[] objects, Queue<JPathToken> tokens)
+        {
+            JPathToken token = DequeueToken(tokens);
+            switch (token.Type)
+            {
+                case JPathTokenType.Asterisk:
+                    if (DequeueToken(tokens).Type != JPathTokenType.RightBracket)
+                        throw new FormatException();
+                    Descent(ref objects);
+                    break;
+                case JPathTokenType.Colon:
+                    ProcessSlice(ref objects, tokens, 0);
+                    break;
+                case JPathTokenType.Number:
+                    JPathToken next = DequeueToken(tokens);
+                    switch (next.Type)
+                    {
+                        case JPathTokenType.Colon:
+                            ProcessSlice(ref objects, tokens, int.Parse(token.Content));
+                            break;
+                        case JPathTokenType.Comma:
+                            ProcessUnion(ref objects, tokens, token);
+                            break;
+                        case JPathTokenType.RightBracket:
+                            Descent(ref objects, int.Parse(token.Content));
+                            break;
+                        default:
+                            throw new FormatException();
+                    }
+                    break;
+                case JPathTokenType.String:
+                    next = DequeueToken(tokens);
+                    switch (next.Type)
+                    {
+                        case JPathTokenType.Comma:
+                            ProcessUnion(ref objects, tokens, token);
+                            break;
+                        case JPathTokenType.RightBracket:
+                            Descent(ref objects, JObject.Parse($"\"{token.Content.Trim('\'')}\"").GetString());
+                            break;
+                        default:
+                            throw new FormatException();
+                    }
+                    break;
+                default:
+                    throw new FormatException();
+            }
+        }
+
+        private static void ProcessRecursiveDescent(ref JObject[] objects, Queue<JPathToken> tokens)
+        {
+            List<JObject> results = new();
+            JPathToken token = DequeueToken(tokens);
+            string identifier = token.Type switch
+            {
+                JPathTokenType.Asterisk => null,
+                JPathTokenType.Identifier => token.Content,
+                _ => throw new FormatException(),
+            };
+            while (objects.Length > 0)
+            {
+                if (identifier is null)
+                    Descent(ref objects);
+                else
+                    Descent(ref objects, identifier);
```

Comment:
I am not sure, this is correct:
For `$..a` and `{"a": 1, "sub": {"a": 2}}` it will return `[1]`, while `[1,2]` is expected.

Reply:
Fixed!